File: 操作系统/Windows API/Windows 程序设计/第6版/CPlusPlus/Chapter11/PetzoldProgrammingWindows6Chapter11/PetzoldProgrammingWindows6Chapter11/ByteToHexStringConverter.cpp
#include "pch.h"
#include "ByteToHexStringConverter.h"

using namespace PetzoldProgrammingWindows6Chapter11;

using namespace Platform;
using namespace Windows::UI::Xaml;
using namespace Windows::UI::Xaml::Interop;

Object^ ByteToHexStringConverter::Convert(Object^ value, TypeName targetType, 
                                          Object^ parameter, String^ langauge)
{
    wchar_t result[10];
    swprintf(result, 10, L"%02X", (unsigned char)value);
    return ref new String(result);
}

Object^ ByteToHexStringConverter::ConvertBack(Object^ value, TypeName targetType, 
                                              Object^ parameter, String^ language)
{
    return value;
}

File: 操作系统/Windows API/Windows 程序设计/第6版/CSharp/Chapter11/Petzold.ProgrammingWindows6.Chapter11/Petzold.ProgrammingWindows6.Chapter11/ByteToHexStringConverter.cs
using System;
using Windows.UI.Xaml.Data;

namespace Petzold.ProgrammingWindows6.Chapter11
{
    public class ByteToHexStringConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            return ((byte)value).ToString("X2");
        }
        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            return value;
        }
    }
}
